src/send/SendSelectRecipient.tsx

```diff
@@ -135,27 +152,167 @@ const getStartedStyles = StyleSheet.create({
+  const recipientInfo = useSelector(recipientInfoSelector)
+
   const contactRecipients = useMemo(
     () => sortRecipients(Object.values(contactsCache)),
     [contactsCache]
   )
+  const [contactsFiltered, setContactsFiltered] = useState(() =>
+    sortRecipients(Object.values(contactRecipients))
+  )
+  const [recentFiltered, setRecentFiltered] = useState(() => recentRecipients)
+
+  const recentRecipientsFilter = useMemo(
+    () => filterRecipientFactory(recentRecipients, false),
+    [recentRecipients]
+  )
+  const contactRecipientsFilter = useMemo(
+    () => filterRecipientFactory(Object.values(contactRecipients), true),
+    [contactRecipients]
+  )
+
+  const getUniqueSearchRecipient = (): Recipient | undefined => {
+    const parsedNumber = parsePhoneNumber(
+      searchQuery,
+      defaultCountryCode ? defaultCountryCode : undefined
+    )
+    if (parsedNumber) {
+      return {
+        displayNumber: parsedNumber.displayNumber,
+        e164PhoneNumber: parsedNumber.e164Number,
+        recipientType: RecipientType.PhoneNumber,
+        name: t('sendToMobileNumber'),
+      }
+    }
+    if (isValidAddress(searchQuery)) {
+      return getRecipientFromAddress(searchQuery.toLowerCase(), recipientInfo)
+    }
+  }
+
+  const mapResolutionToRecipient = (resolution: NameResolution): Recipient => {
+    const lowerCaseAddress = resolution.address.toLowerCase()
+    switch (resolution.kind) {
+      case ResolutionKind.Address:
+        return getRecipientFromAddress(lowerCaseAddress, recipientInfo)
+      case ResolutionKind.Nom:
+        return {
+          address: lowerCaseAddress,
+          name: t('nomSpaceRecipient', { name: resolution.name ?? searchQuery }),
+          recipientType: RecipientType.Nomspace,
+        }
+      default:
+        return getRecipientFromAddress(lowerCaseAddress, recipientInfo)
+    }
+  }
+
+  const mergedFilteredRecipients = useMemo(() => {
+    const allRecipients: Recipient[] = []
+    if (resolveAddressResult && resolveAddressResult.resolutions.length > 0) {
+      const resolutions = resolveAddressResult.resolutions.map(mapResolutionToRecipient)
+      allRecipients.push(...resolutions)
+    }
+    allRecipients.push(...recentFiltered)
+    allRecipients.push(...contactsFiltered)
+
+    const mergedRecipients: Recipient[] = []
+    for (const potentialRecipient of allRecipients) {
+      if (
+        !mergedRecipients.find(
+          (mergedRecipient) =>
+            mergedRecipient.e164PhoneNumber === potentialRecipient.e164PhoneNumber ||
+            mergedRecipient.address === potentialRecipient.address
+        )
+      ) {
+        mergedRecipients.push(potentialRecipient)
+      }
+    }
+
+    const uniqueRecipient = getUniqueSearchRecipient()
+    if (!mergedRecipients.length && uniqueRecipient) {
+      mergedRecipients.push(uniqueRecipient)
+    }
+
+    return mergedRecipients
+  }, [recentFiltered, contactsFiltered, resolveAddressResult])
```

Comment:
uniqueRecipient could also change whenever searchQuery changes

Reply:
Ah yeah, that's a good point -- I think without this, we could run into a situation where this doesn't update when it actually should.